### PR TITLE
Improve arm64 esil

### PR DESCRIPTION
**Pull request by mrmacete**
- add implementations for MSUB, BFI, NEG
- fix BLR implementation
- move "tmp" register to avoid overlapping "sp" in register profile

Tests here: https://github.com/radare/radare2-regressions/pull/822

### CODE REVIEW

File: libr/anal/p/anal_arm_cs.c [hunk 1]
```diff
@@ -1254,6 +1257,34 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
...
+		// notmask,dst,&,lsb,mask,src,&,<<,|,dst,=
+
+		r_strbuf_setf (&op->esil, "%"PFMT64u",%s,&,%"PFMT64u",%"PFMT64u",%s,&,<<,|,%s,=",
+			notmask,
+			REG64(0),
+			shift,
+			mask,
+			REG64(1),
+			REG64(0));
+
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: remove empty lines

File: libr/anal/p/anal_arm_cs.c [hunk 2]
```diff
@@ -1254,6 +1257,34 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
...
+	}
+	case ARM64_INS_NEG:
+	case ARM64_INS_NEGS:
+		if (LSHIFT2_64 (1)) {
+			SHIFTED_REG64_APPEND (&op->esil, 1);
+		} else {
+			r_strbuf_appendf (&op->esil, "%s", REG64(1));
+		}
+		r_strbuf_appendf (&op->esil, ",0,-,%s,=", REG64(0));
+
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: empty line

File: libr/anal/p/anal_arm_cs.c [hunk 3]
```diff
@@ -890,6 +890,10 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
 		r_strbuf_setf (&op->esil,
 			"%s,%s,*,%s,+,%s,=",REG64(2),REG64(1),REG64(3), REG64(0));
 		break;
+	case ARM64_INS_MSUB:
+		r_strbuf_setf (&op->esil,
+			"%s,%s,*,%s,-,%s,=",REG64(2),REG64(1),REG64(3), REG64(0));
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: add spaces after commas, maybe the format string fits in the line above and will be more readable

File: libr/anal/p/anal_arm_cs.c [hunk 4]
```diff
@@ -1254,6 +1257,34 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
 	case ARM64_INS_RET:
 		r_strbuf_setf (&op->esil, "lr,pc,=");
 		break;
+	case ARM64_INS_BFI: // bfi w8, w8, 2, 1
+	case ARM64_INS_BFXIL:
+	{
+		ut64 mask = bitmask_by_width[IMM64(3)-1];
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: missing space before (

Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: and around -

File: libr/anal/p/anal_arm_cs.c [hunk 5]
```diff
@@ -1254,6 +1257,34 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
...
+			REG64(0));
+
+		break;
+	}
+	case ARM64_INS_NEG:
+	case ARM64_INS_NEGS:
+		if (LSHIFT2_64 (1)) {
+			SHIFTED_REG64_APPEND (&op->esil, 1);
+		} else {
+			r_strbuf_appendf (&op->esil, "%s", REG64(1));
```
Comment (radare) [inline libr/anal/p/anal_arm_cs.c]: missing space before (

File: libr/anal/p/anal_arm_cs.c (hunk 1 above)
Comment (XVilka) [inline libr/anal/p/anal_arm_cs.c]: may be fold this column in a line instead?

**Pull request merged**